opencti-platform/opencti-front/src/private/components/common/form/StatusTemplateField.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { pathOr, pipe, map, union, append } from 'ramda';
+import { Field } from 'formik';
+import { Label } from 'mdi-material-ui';
+import makeStyles from '@mui/styles/makeStyles';
+import { fetchQuery } from '../../../../relay/environment';
+import AutocompleteField from '../../../../components/AutocompleteField';
+import { statusTemplatesSearchQuery } from '../../settings/StatusTemplatesQuery';
+import StatusTemplateCreation from '../../settings/workflow/StatusTemplateCreation';
+import { useFormatter } from '../../../../components/i18n';
+
+const useStyles = makeStyles(() => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+  autoCompleteIndicator: {
+    display: 'none',
+  },
+}));
+
+const StatusTemplateField = ({ name, setFieldValue, helpertext }) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const [statusTemplateCreation, setStatusTemplateCreation] = useState(false);
+  const [statusTemplateInput, setStatusTemplateInput] = useState('');
+  const [statusTemplates, setStatusTemplates] = useState([]);
+
+  const handleOpenStatusTemplateCreation = () => setStatusTemplateCreation(true);
+
+  const handleCloseStatusTemplateCreation = () => setStatusTemplateCreation(false);
+
+  const searchStatusTemplates = (event) => {
+    setStatusTemplateInput(event && event.target.value !== 0 ? event.target.value : '');
+    fetchQuery(statusTemplatesSearchQuery, {
+      search: event && event.target.value !== 0 ? event.target.value : '',
+    })
+      .toPromise()
+      .then((data) => {
+        const NewStatusTemplates = pipe(
+          pathOr([], ['statusTemplates', 'edges']),
```

Comment:
Can we remove ramda by using javascript functions like statusTemplates?.edges?.map() ?